### PR TITLE
Fix geolocate control permissions failure on IOS16 web view

### PR DESCRIPTION
**Pull request by richturner**
## Launch Checklist

On iOS16 (and possibly other environments) web view `window.navigator.permissions.query({name: 'geolocation'})` is rejected and the error was never handled.

This PR includes a small update to catch the exception and fallback on the `window.navigator.geolocation` API.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ### ✨ Features and improvements
 - *...Add new stuff here...*
+- Fix geolocate control permissions failure on IOS16 web view with fallback to `window.navigator.geolocation`
```
Comment (HarelM) [inline CHANGELOG.md:5]: I would expect to see this under bugs...?

File: src/ui/control/geolocate_control.ts [hunk 2]
```diff
@@ -44,6 +44,10 @@ function checkGeolocationSupport(callback) {
         window.navigator.permissions.query({name: 'geolocation'}).then((p) => {
             supportsGeolocation = p.state !== 'denied';
             callback(supportsGeolocation);
+        }).catch(() => {
+            // Fix for iOS16 which rejects query but still supports geolocation
+            supportsGeolocation = !!window.navigator.geolocation;
```
Comment (HarelM) [inline src/ui/control/geolocate_control.ts:49]: What happens if the user rejects? This seems wrong to me, but I haven't tested it.

**Comment by HarelM** [comment]
Thanks for taking the time to open this PR!
Can you please add a browser test to check this functionality somehow? If not, at least a unit test.
Thanks!

File: src/ui/control/geolocate_control.ts (hunk 2 above)
Comment (richturner) [reply src/ui/control/geolocate_control.ts:49]: `window.navigator.permissions.query` doesn't request permission it checks the status of the permission

Comment (HarelM) [reply src/ui/control/geolocate_control.ts:49]: I see, so in what cases other than iOS 16 this can throw a excretion?

Comment (HarelM) [reply src/ui/control/geolocate_control.ts:49]: Also, can we use a promise here instead of a callback?

Comment (richturner) [reply src/ui/control/geolocate_control.ts:49]: I don't know if other environments cause an error to be thrown but only know that this is causing us problems in our web app on iOS16.

I just put a fix in following existing code structure feel free to make bigger adjustments

**Review by HarelM — APPROVED** [review]

**Comment by HarelM** [comment]
@richturner any updates regarding a test for this PR?

**Comment by HarelM** [comment]
Superseded by #2359

**Pull request closed without merging**